client/fingerprint/env_aws.go

```diff
@@ -116,6 +116,9 @@ func (f *EnvAWSFingerprint) Fingerprint(cfg *config.Config, node *structs.Node)
 	}
 	for _, k := range keys {
 		res, err := client.Get(metadataURL + k)
+		if res.StatusCode != 200 {
```

Comment:
I would probably log something here with WARN, just so people can see why some of AWS metadata is not showing up upon `nomad node-status`